master/master.cfg

```diff
@@ -46,32 +46,52 @@ WORKER_SECRET = Path('halide_bb_pass.txt').read_text().strip()
 WEBHOOK_SECRET = Path('webhook_token.txt').read_text().strip()
 WWW_PASSWORD = Path('buildbot_www_pass.txt').read_text().strip()
 
+# SERVER SETTINGS
+
+ARTIFACTS_DIR = os.environ.get('HALIDE_BB_MASTER_ARTIFACTS_DIR', '/home/abadams/artifacts')
+REPO_DIR = Path(__file__, '..', '..').resolve()
+
 # LLVM
 
 # At any given time, we test (at least) 3 LLVM versions:
 # - the current trunk (changes daily)
 # - the most recent release (expected to be stable)
 # - an older release (expected to be stable)
 #
-# the branches that correspond to these will rotate as new versions
-# are released, but the underlying test logic should not need changing
+# The branches that correspond to these will rotate as new versions
+# are released, but the underlying test logic should not need changing.
 
-LLVM_TRUNK_BRANCH = 'main'
-LLVM_RELEASE_BRANCH = 'release/11.x'
-LLVM_OLD_BRANCH = 'release/10.x'
+Version = namedtuple('Version', ['major', 'minor', 'patch'])
+VersionedBranch = namedtuple('VersionedBranch', ['ref', 'version'])
 
-_LLVM_BRANCHES = [LLVM_TRUNK_BRANCH, LLVM_RELEASE_BRANCH, LLVM_OLD_BRANCH]
+LLVM_TRUNK = 'trunk'
+LLVM_RELEASE_11 = 'release_11'
+LLVM_RELEASE_10 = 'release_10'
 
-_TO_VERSION = {
-    LLVM_TRUNK_BRANCH: 12,
-    LLVM_RELEASE_BRANCH: 11,
-    LLVM_OLD_BRANCH: 10,
-}
+LLVM = {LLVM_TRUNK: VersionedBranch(ref='main', version=Version(12, 0, 0)),
+        LLVM_RELEASE_11: VersionedBranch(ref='release/11.x', version=Version(11, 0, 1)),
+        LLVM_RELEASE_10: VersionedBranch(ref='release/10.x', version=Version(10, 0, 1))}
+
+# At any given time, Halide has a trunk branch, which supports (at least)
+# the LLVM trunk branch and the most recent release branch (and maybe one older).
+#
+# We also support previous release branches; a release branch tracks *only* the
+# corresponding version of LLVM (i.e., Halide 11 is 'release/11.x' and is only
+# built/tested against LLVM11, even though it might still work with other LLVM versions).
+#
+# Note that we deliberately chose branch names that match LLVM's conventions.
+#
+# (Note that there is a 'release/8.x' branch of Halide but we are going to ignore it here.)
 
+HALIDE_TRUNK = 'trunk'
```

Comment:
note to self: as a followup, standalone PR, let's do a name cleanup that changes all use of 'trunk' and 'master' to be 'main' instead (except where still necessary for actual git branch names)

Reply:
Good idea